torchaudio/functional.py

```diff
@@ -524,34 +524,38 @@ def lfilter(waveform, a_coeffs, b_coeffs):
 
     Returns:
         output_waveform (torch.Tensor): Dimension of `(n_channel, n_frames)`.  Output will be clipped to -1 to 1.
+                                        Will be on the same device as the inputs.
 
     """
 
     assert(waveform.dtype == torch.float32)
```

Comment:
Why do we limit to `float32`? It might be a better idea to pass the `dtype` to the different computations.

Reply:
Same below. By the way, let's be careful when we create other tensors to :)

Reply:
In a similar vein, we should probably avoid using the math library were we can use pytorch, e.g. [here](https://github.com/pytorch/audio/blob/c9926814cd57fa8dc176fe1387ff05eb60fef5fe/torchaudio/functional.py#L599).

Reply:
I'd prefer not to cast to a tensor just to do an exp.  This is done very few times, just in the filter design portion.  Also `math` is a python standard library.  https://pytorch.org/docs/stable/torch.html?highlight=exp#torch.exp

Reply:
I see, they are used only for scalars. I agree that's fine.